### ISSUE TITLE
Docker container not not serve assets when a prefix path is configured 

### ISSUE THREAD

**Issue by johannes-darms** [seek4science/seek#796]
**Describe the bug**
While running the plain docker-compose installation, behind a reverse-proxy with a prefix path (subURL similar to this [documentation](https://docs.seek4science.org/tech/install-on-suburi.html)) assets are not correctly displayed. Probability the issue can be resolved by executing `RAILS_ENV=production bundle exec rake assets:precompile` within the container. However,    this is a manual step needed and conflicts with the idea and concept that a container can be easily, without manual work, recreated from an image. This hinder the deployment of SEEK within k8s environments.
Hence, I propose to change the docker image so that, the docker environment variable "RAILS_RELATIVE_URL_ROOT" is considered and assets are correctly provided.

**Comment by johannes-darms** [comment]
@whomingbird: Could you raise this issue in one of your developer meetings. This change would simplify the deployment quite a bit.

**Comment by stuzart** [comment]
Yes this is something I've thought about for some time, and would be good to fix. You're right that the `RAILS_ENV=production bundle exec rake assets:precompile` would need to be re-run using the environment variable. Because the assets and bundled within the container image, this would need to happen each time the container is recreated, i.e following a docker-compose down, which could cause a bottleneck - but it would only need to do so if `RAILS_RELATIVE_URL_ROOT` has been set.

**Comment by stuzart** [comment]
... could also include an "optional" volume that allows the assets in this case to be persisted outside the container. Then each time the precompile is re-run at start-up it should be very quick if has already previously processed them.

**Comment by johannes-darms** [comment]
This would be to update to `entrypoint.sh` to run the precompile step every time a container is started and the environment variable `RAILS_RELATIVE_URL_ROOT` is set. This delays the startup, but would solve the problem without much effort. Do you know how much time is spend on this task?  Unfortunately, I have never worked with Rails, but my gut tells me there is probably a smarter solution.
I wouldn't use a dedicated volume, as they often outlast the container, which can lead to an inconsistent state.

**Comment by stuzart** [comment]
I haven't tested it, but I think https://github.com/seek4science/seek/commit/54e3e0260807ee75c3720945b20a76fad879a371 should do it. Along with adding the RAILS_RELATIVE_URL_ROOT to the environment for `seek` in the `docker-compose.yml` . 

@whomingbird could try this on her branch that your container gets built from.

( I'm aware that "-n" may be better than "! -z" but I've left it be the same as the other cases for now )

I do think the volume may be needed to prevent long start up times, but you can give it a try without first. Including `rake assets:clean` could be used to avoid an inconsistent state.

**Comment by johannes-darms** [comment]
Works like charm, thanks! However, the initial startup time increased by ~4 minutes.

**Comment by stuzart** [comment]
Great, and thanks for your help being our guinea pig :)

The additional 4 minute startup would drive me mad, so I tried out using a volume and it seems to work well. I also created an example docker-compose, along with adding --trace to the precompile step so it doesn't appear like everything has hung. It is not necessary to use the volume if you don't need to, so shouldn't need to change anything. 
The changes can be found in https://github.com/seek4science/seek/commit/35c548868a2aeb098eed9c2ac6c3029263d20045
